pkg/rehearse/jobs.go

```diff
@@ -266,37 +263,78 @@ func getResolvedConfigForTest(ciopConfigs config.DataByFilename, resolver regist
+			if err := yaml.Unmarshal([]byte(env.Value), &ciopConfig.Configuration); err != nil {
+				return nil, fmt.Errorf("failed to unmarshal UNRESOLVED_CONFIG: %w", err)
+			}
+			// Annoying hack: UNRESOLVED_CONFIG means this is a handcrafted job, which means
+			// `testname` cannot be relied on (it is derived from job name, which is arbitrary
+			// in handcrafted jobs). We need the test name to know which `tests` field to
+			// resolve, so we try to detect it from `--target` arg, if present.
+			//
+			// The worst case is that we do not find the matching name. In such case,
+			// the inlined config will have no `tests` stanza, which may or may not be
+			// the right thing, but in such case the rehearsal behavior is identical
+			// to the production job.
+			for idx, arg := range container.Args {
+				if strings.HasPrefix(arg, "--target=") {
+					testname = strings.TrimPrefix(arg, "--target=")
+					break
+				}
+				if arg == "--target" {
+					if len(container.Args) == (idx + 1) {
+						return nil, errors.New("plain '--target' is a last arg, expected to be followed with a value")
+					}
+					testname = container.Args[idx+1]
+					break
+				}
+			}
+			hasConfigEnv = true
+		default:
+			// Another envvar, we just need to keep it
+			envs = append(envs, container.Env[idx])
 		}
 	}
 
-	// inline CONFIG_SPEC for all ci-operator jobs
-	if container.Command != nil && container.Command[0] == "ci-operator" {
+	if !hasConfigEnv {
 		if err := metadata.IsComplete(); err != nil {
 			return nil, fmt.Errorf("could not infer which ci-operator config this job uses: %w", err)
 		}
 		filename := metadata.Basename()
-		loggers.Debug.WithField(logCiopConfigFile, filename).Debug("Rehearsal job uses ci-operator config ConfigMap, needed content will be inlined")
-		ciOpConfigContent, imageStreamTags, err := getResolvedConfigForTest(ciopConfigs, resolver, filename, testname)
-		if err != nil {
-			loggers.Job.WithError(err).Error("Failed to get resolved config for test")
-			return nil, err
+		if _, ok := ciopConfigs[filename]; !ok {
+			return nil, fmt.Errorf("ci-operator config file %s was not found", filename)
 		}
-		apihelper.MergeImageStreamTagMaps(allImageStreamTags, imageStreamTags)
+		ciopConfig = ciopConfigs[filename]
+		loggers.Debug.WithField(logCiopConfigFile, filename).Debug("Rehearsal job would use ci-operator config from registry, its content will be inlined")
+	}
 
-		envs := container.Env
-		env := v1.EnvVar{
-			Name:  "CONFIG_SPEC",
-			Value: ciOpConfigContent,
-		}
-		envs = append(envs, env)
-		container.Env = envs
+	ciOpConfigContent, imageStreamTags, err := getResolvedConfigForTest(ciopConfig, resolver, testname)
+	if err != nil {
+		loggers.Job.WithError(err).Error("Failed to get resolved config for test")
+		return nil, err
 	}
+	apihelper.MergeImageStreamTagMaps(allImageStreamTags, imageStreamTags)
+	container.Env = append(envs, v1.EnvVar{Name: "CONFIG_SPEC", Value: ciOpConfigContent})
 	return allImageStreamTags, nil
```

Comment:
Could we update the godoc to explain why a method that is supposed to in-line config returns a map of imagestreamtags?

Reply:
heh, I'll need to figure out that information myself, but yeah